lib/host/wasi/inode-win.cpp

```diff
@@ -489,8 +489,24 @@ WasiExpect<INode> INode::open(std::string Path, __wasi_oflags_t OpenFlags,
   DWORD AccessFlags = accessFlags(FdFlags, VFSFlags);
   DWORD CreationDisposition = creationDisposition(OpenFlags);
 
+  int NumCharacters =
+      MultiByteToWideChar(CP_ACP, MB_PRECOMPOSED, Path.c_str(), -1, nullptr, 0);
+
+  if (unlikely(NumCharacters <= 0)) {
+    return WasiUnexpect(fromWinError(GetLastError()));
+  }
+
+  WCHAR PathBuffer[2 * NumCharacters];
```

Comment:
Don't use VLA in C++. Ref: https://stackoverflow.com/questions/1887097/why-arent-variable-length-arrays-part-of-the-c-standard

Reply:
I will use `MAX_PATH` instead and also will remove the multiplication by 2 - leftover from the previous attempt :)

Reply:
AFAIK the limitation of `MAX_PATH` can be removed after Windows 10 build 1607 by using the following commands:

```PowerShell
New-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem" `
-Name "LongPathsEnabled" -Value 1 -PropertyType DWORD -Force
```

Maybe we will need to check the `NumCharacters` if it's larger than the `MAX_PATH.`

Ref: https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

Reply:
@hydai Soory for the late reply.

> Maybe we will need to check the `NumCharacters` if it's larger than the `MAX_PATH.`

I think that you are suggesting this: We should check if the `Path` is longer than `MAX_PATH` and if it is then we should query the registry checking if long paths are enabled and if they are enabled then return an error else continue with the normal execution.

I think that this approach is too cumbersome and I suggest that we should do this instead:
Define a constant with the value `32,767` which will be the size of the buffer- `PathBuffer`. So we don't have to worry about the path lengths as `CreateFileW` allows for long paths by default.